node/src/blockchain/blockchain_interface/blockchain_interface_web3/lower_level_interface_web3.rs

```diff
@@ -1,395 +1,762 @@
-            REQUESTS_IN_PARALLEL,
-        )
-        .unwrap();
-        let chain = TEST_DEFAULT_CHAIN;
-        let subject = make_subject(transport, chain);
+        let _blockchain_client_server = MBCSBuilder::new(port)
+            .response("0x23".to_string(), 1)
+            .start();
+        let subject = make_blockchain_interface_web3(Some(port));
+        let wallet = &Wallet::from_str("0x3f69f9efd4f2592fd70be8c32ecd9dce71c472fc").unwrap();
+
+        let result = subject
+            .lower_interface()
+            .get_transaction_id(wallet.address())
+            .wait();
+
+        assert_eq!(result, Ok(35.into()));
+    }
+
+    #[test]
+    fn get_transaction_id_returns_an_error_for_unintelligible_response() {
+        let port = find_free_port();
+        let _blockchain_client_server = MBCSBuilder::new(port)
+            .response("0xFFFQ".to_string(), 0)
+            .start();
+        let subject = make_blockchain_interface_web3(Some(port));
 
         let result = subject
+            .lower_interface()
+            .get_transaction_id(
+                Wallet::from_str("0x3f69f9efd4f2592fd70be8c32ecd9dce71c472fc")
+                    .unwrap()
+                    .address(),
+            )
+            .wait();
+
+        match result {
+            Err(BlockchainError::QueryFailed(msg)) if msg.contains("invalid hex character: Q") => {
+                ()
+            }
+            x => panic!("Expected complaint about hex character, but got {:?}", x),
+        };
+    }
+
+    #[test]
+    fn get_token_balance_can_retrieve_token_balance_of_a_wallet() {
+        let port = find_free_port();
+        let _blockchain_client_server = MBCSBuilder::new(port)
+            .response(
+                "0x000000000000000000000000000000000000000000000000000000000000FFFF".to_string(),
+                0,
+            )
+            .start();
+        let subject = make_blockchain_interface_web3(Some(port));
+
+        let result = subject
+            .lower_interface()
             .get_service_fee_balance(
-                &Wallet::from_str("0x3f69f9efd4f2592fd70be8c32ecd9dce71c472fc").unwrap(),
+                Wallet::from_str("0x3f69f9efd4f2592fd70be8c32ecd9dce71c472fc")
+                    .unwrap()
+                    .address(),
             )
+            .wait()
             .unwrap();
 
-        assert_eq!(result, U256::from(0xDEADBEEF_u64));
-        let requests = test_server.requests_so_far();
-        let bodies: Vec<Value> = requests
-            .into_iter()
-            .map(|request| serde_json::from_slice(&request.body()).unwrap())
-            .collect();
-        assert_eq!(bodies[0]["method"].to_string(), "\"eth_call\"",);
-        let contract_address = chain.rec().contract;
-        assert_eq!(
-            bodies[0]["params"][0]["to"].to_string(),
-            format!("\"{:?}\"", contract_address),
-        );
-        assert_eq!(bodies.len(), 1)
+        assert_eq!(result, U256::from(65_535));
```

Comment:
Same as previously mentioned (again two things, one is an optimization another is a big issue)